common/models/user.js

```diff
@@ -360,21 +367,34 @@ module.exports = function(User) {
     // Email model
     var Email = options.mailer || this.constructor.email || loopback.getModelByType(loopback.Email);
 
-    crypto.randomBytes(64, function(err, buf) {
-      if (err) {
-        fn(err);
-      } else {
-        user.verificationToken = buf.toString('hex');
-        user.save(function(err) {
-          if (err) {
-            fn(err);
-          } else {
-            sendEmail(user);
-          }
-        });
-      }
+    // Set a default token generation function if one is nto provided
+    var tokenGenerator = options.tokenGenerator || defaultTokenGenerator;
+
+    tokenGenerator(user, function(token) {
+      user.verificationToken = token;
+      user.save(function(err) {
+        if (err) {
+          fn(err);
+        } else {
+          sendEmail(user);
+        }
+      });
     });
 
+    /**
+     * A default verification token generator which accepts a user and callback.
+     * This one uses the crypto library and 64 random bytes (converted to hex)
+     * for the token.
+     *
+     * @options {Object} user The user this token is being generated for
+     * @options {Function} tokenCallback The generator will pass back the new token with this function call
+     */
+    function defaultTokenGenerator(user, tokenCallback) {
```

Comment:
I think my point about putting it on the `prototype` was that you can *only* override it once. A developer could not have two different implementations within the same application at the same time. Again, if we really need this to be on the prototype we can, but I think that limits the functionality.

But to the other point, I was not thinking that this method would apply to _all_ token generation, only for `verify`. I think I see where you are going with the functionality, but that is a bigger overhaul, correct? That would mean centralizing all token generation, which is not how the code is currently set up.